instaloader/instaloadercontext.py

```diff
@@ -47,15 +47,30 @@ class InstaloaderContext:
     class :class:`Instaloader`.
     """
 
-    def __init__(self, sleep: bool = True, quiet: bool = False, user_agent: Optional[str] = None,
-                 graphql_count_per_slidingwindow: Optional[int] = None, max_connection_attempts: int = 3):
+    def __init__(
+            self,
+            sleep: bool = True,
+            quiet: bool = False,
+            user_agent: Optional[str] = None,
+            graphql_count_per_slidingwindow: Optional[int] = None,
+            max_connection_attempts: int = 3,
+            proxy: Optional[str] = None,
+    ):
 
         self.user_agent = user_agent if user_agent is not None else default_user_agent()
         self._session = self.get_anonymous_session()
```

Comment:
Did you consider setting the `proxies` dictionary in the requests.Session object, rather than passing it to each Session.get call? I think it might be the better option since there are fewer placer where sessions are created than where they are used to make a request.

http://docs.python-requests.org/en/master/api/#requests.Session.proxies